### PR TITLE
Create MUI v4-compatible theme

### PR DESCRIPTION
**Pull request by joebochill**
For PXBLUE-1078.

Changes proposed in this Pull Request:
- Update React theme to MUI4 spec
- Removed unused props
- Cleanup shared folder
- Update copyright dates

### CODE REVIEW

**Comment by huayunh** [comment]
change log?

**Comment by emclaug2** [comment]
Will this theme upgrade still work with Material UI v3?  Not sure if the README needs updated.

**Comment by joebochill** [comment]
> Will this theme upgrade still work with Material UI v3? Not sure if the README needs updated.

In theory, it could _work_, but there will likely be issues and things some things will likely not look the way we want. I would prefer to say this theme is for Material v4 and up.

**Comment by joebochill** [comment]
> How frequently should we update the docit roadmap? Per sprint?

I would update it as things change for now until we find a rhythm/cadence that makes sense. Ideally we can do the story to load the data from a database and then it will be a much simpler process to update in near real-time

File: react/src/blueDarkTheme.ts [hunk 1]
```diff
@@ -1,147 +1,151 @@
...
-                    color: ThemeColors.black['50'],
-                    backgroundColor: ThemeColors.black['A700']
-                },
-                colorSecondary:{
-                    backgroundColor: ThemeColors.black['A100']
-                }
+        text: {
+            primary: ThemeColors.gray['300'],
+            secondary: ThemeColors.black['300'],
+            hint: ThemeColors.gray['300'],
```
[the remark this answers is not in the document]
Comment (joebochill) [reply react/src/blueDarkTheme.ts:30]: `theme.palette.text.hint` is not used by any MUI component or by us at the moment

File: react/src/shared.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+export const typography = {
+  fontFamily: '"Open Sans", Helvetica, Roboto, sans-serif',
+  fontWeightMedium: 600,
+};
+export const createSimplePalette = (color: PXBlueColor): SimplePaletteColorOptions => ({
+  light: color[50],
+  main: color[500],
+  dark: color[700],
+});
```
[the remark this answers is not in the document]
Comment (joebochill) [reply react/src/shared.ts:20]: prettier is not integrated into the themes project

[the remark this answers is not in the document]
Comment (joebochill) [reply react/src/shared.ts:20]: I can add it if we want

File: react/src/blueDarkTheme.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (huayunh) [reply react/src/blueDarkTheme.ts:29]: We will have... soon...

I think there are plenty of gray500 in our react components. You might need to doublecheck and ensure that they are using the latest theme package.

[the remark this answers is not in the document]
Comment (joebochill) [reply react/src/blueDarkTheme.ts:29]: we're definitely using gray[500] in places. But there's not a single reference to `theme.palette.text.hint` in any MUI component or PX Blue component right now. So changing this value will have no impact right now. Unless there's upcoming components that are expected to use it, which I can't think of at least for the next several months.

**Pull request merged**